### DISCUSSION TITLE
HDF5 file explorer that integrates in lf

### DISCUSSION THREAD

**Discussion by jpmvferreira** [gokcehan/lf#1602] [Q&A]
Greetings everyone, I've been searching for command line tools that makes exploring an HDF5 file similar to what one does with directories.
Given the hierarchical nature of HDF5, this isn't completely unreasonable, as you would act as if groups were folders, the metadata associated with it as text files that could be displayed with a preview, as lf allows it to, and then the datasets would be ordinary files.

I would like to ask you then:
1. Are you aware of any tool that turns HDF5 files into a browsable folder?
2. Can such a tool be integrated nicely in lf?

Many thanks!

**Comment by ilyagr** [comment]
I found https://github.com/zjttoefs/hdfuse5 and a few others by searching for "hdf5 fuse". They all seem to be proofs of concepts rather than seriously supported tools, though. They are probably Linux only at the moment, though I know there are tools to adapt FUSE-based programs to run on Windows (dokany, maybe?). Mac OS is weirder, it's getting more and more difficult to install MacFUSE, but there's also FUSE-T.

**Reply by jpmvferreira** [reply]
That's also a clever idea, although it would require more steps to integrate with lf, but I don't see how could one have lf interpret a file in the same way as a folder if it can be browsable, e.g. I also a vim plugin to browse tar files, but I would also like to have that integrated in lf somehow.
